### PR TITLE
docs: split building and deploying doc

### PR DESCRIPTION
**Pull request by skoh7645**
## Checklist

- [x] commit message follows [commit guidelines](https://github.com/appsody/website/blob/master/CONTRIBUTING.md#commit-message-guidelines).
- [x] website builds and runs in production - for information on how to test locally go [here](https://github.com/appsody/website/blob/master/DEVELOPMENT.md).

## Description
Split up the "building and deploying" doc into " building", "deploying" and "appsody operator" docs. I tried to keep the content largely the same in this PR. Future PRs will be made to improve the actual content. 

For easier reviewing:
- [ ] "Building" doc is taken from the original "building and deploying" doc. I only added the first 2 paragraphs and the last line
- [ ] "Deploying" doc is the original "building and deploying" doc with the "building" and "appsody operator" sections removed from it. I only made changes to the `# Getting ready to deploy your Appsody project` section, and the last sentence in the `### Deployment via the Appsody Operator` section
- [ ] "Appsody Operator" doc is directly taken from the original "building and deploying" doc with no additional changes

## Related Issues
 Related to https://github.com/appsody/website/issues/359

### CODE REVIEW

File: content/docs/reference/appsody-operator.md [hunk 1]
```diff
@@ -0,0 +1,122 @@
+---
+title: Appsody Operator
+---
+
+# How the Appsody operator works
+Before we delve into the details of `appsody deploy` and `appsody operator`, we need to spend a few words on how the Appsody operator works.
```
Comment (helenmasters) [inline content/docs/reference/appsody-operator.md:6]: This seems like a strange place to start a page.... "Before we delve into...?" I think given this is now on its own page you can lose this whole sentence.

**Comment by helenmasters** [comment]
You'll need to add redirects, changes to the sidebar yaml and any direct website refs also need updating...

File: content/docs/using-appsody/building.md [hunk 2]
```diff
@@ -0,0 +1,37 @@
+---
+title: Building your Appsody project
+---
+
+# Building your Appsody project
+
+The Appsody CLI downloads and runs a *development* Docker image of the target runtime during local development. This image differs slightly from the image that is used at deployment time, because it configures tools that are useful only during the development phase.
+
+The Appsody CLI provides the [`appsody build`](/docs/cli-commands/#appsody-build) command to generate a *deployment* Docker image on your local Docker registry.
```
Comment (helenmasters) [inline content/docs/using-appsody/building.md:9]: It feels like this should be the first sentence and the sentence/paragraph above should be a note/comment underneath, as it seems strange to talk about development first on a page about build.

File: content/docs/using-appsody/deploying.md [hunk 3]
```diff
@@ -0,0 +1,112 @@
+---
+title: Getting ready to deploy your Appsody project
+---
+
+# Getting ready to deploy your Appsody project
+When you've finished the development work for your Appsody project, you will have a containerized application that's ready to deploy to a suitable runtime infrastructure such as a cloud platform that hosts a Kubernetes cluster.
+
+The Appsody CLI provides the `appsody deploy` command to build and deploy a Docker image directly to a Kubernetes cluster that you are using for testing or staging.
+
+You can delegate the build and deployment steps to an external pipeline, such as a Tekton pipeline that consumes the source code of your Appsody project after you push it to a GitHub repository. Within the pipeline, you can run `appsody build`, which builds the application image and generates a deployment manifest. You can use the manifest to deploy your application to a Kubernetes environment where the Appsody operator is installed.
```
Comment (helenmasters) [inline content/docs/using-appsody/deploying.md:10]: You should probably link back to the build page where you reference `appsody build`.

File: content/docs/using-appsody/building.md (hunk 2 above)
Comment (skoh7645) [reply content/docs/using-appsody/building.md]: I've tried to reword it taking this into account :)

**Comment by skoh7645** [comment]
I have added one redirect which is the main one. I think we can add redirects for anchor links that we are aware have been used externally. However, we will probably have to accept some breakages as we can't redirect every anchor header. I've also updated the sidebar yaml and any  direct website link references

**Comment by helenmasters** [comment]
Sandy, I dreamt about the docs last night (sad I know!). Is there a link from the deploying doc to the operator docs as I didn't notice one yesterday? There should be something to show users that there is more information in Reference otherwise we can't expect users to find it.

**Comment by skoh7645** [comment]
@helenmasters haha sorry to hear about that. There is a link to the operator doc at the end of the `### Deployment via the Appsody Operator` section

**Pull request merged**